plugins/inputs/nomad/nomad_test.go

```diff
@@ -0,0 +1,127 @@
+package nomad
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNomadStats(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.RequestURI == "/v1/metrics" {
+			w.WriteHeader(http.StatusOK)
+			_, err := fmt.Fprintln(w, responseKeyMetrics)
+			require.NoError(t, err)
+		}
+	}))
+	defer ts.Close()
+
+	labelFilter, _ := filter.NewIncludeExcludeFilter([]string{"host"}, nil)
+
+	n := &Nomad{
+		URL:         ts.URL,
+		labelFilter: labelFilter,
+	}
+
+	var acc testutil.Accumulator
+	err := acc.GatherError(n.Gather)
+	require.NoError(t, err)
+
+	fields := map[string]interface{}{
+		"value": float32(500),
+	}
+	tags := map[string]string{
+		"node_scheduling_eligibility": "eligible",
+		"host":                        "node1",
+		"node_id":                     "2bbff078-8473-a9de-6c5e-42b4e053e12f",
+		"datacenter":                  "dc1",
+		"node_class":                  "none",
+		"node_status":                 "ready",
+	}
+	acc.AssertContainsTaggedFields(t, "nomad.client.allocated.cpu", fields, tags)
```

Comment:
You can construct an expected metric and the use `testutil.RequireMetricsEqual`. You might take a look at [this example](https://github.com/influxdata/telegraf/blob/b9e4978b17754faa5900e65f53fac4693da1ec13/plugins/processors/starlark/starlark_test.go#L261)...

Reply:
I've looked at the example and I'll rewrite all the tests file entirely